### ISSUE TITLE
Edge case for `lsm_p_core()`

### ISSUE THREAD

**Issue by Nowosad** [r-spatialecology/landscapemetrics#68]
`lsm_p_core()` gives an error when the input raster is just one patch. What's the proper output value here? Zero?

``` r
library(raster)
#> Loading required package: sp
library(landscapemetrics)
x = raster(nrows = 100, ncols = 100, 
           xmn = 0, xmx = 100,
           ymn = 0, ymx = 100,
           crs = NA)
x[] = 2
plot(x)
```

![](https://i.imgur.com/vGClD3q.png)

``` r

# works -------------------------------------------------------------------
lsm_l_area_mn(x)
#> # A tibble: 1 x 6
#>   layer level     class    id metric  value
#>   <int> <chr>     <int> <int> <chr>   <dbl>
#> 1     1 landscape    NA    NA area_mn     1

# fails -------------------------------------------------------------------
lsm_p_core(x)
#> Error in cells_patch - cells_edge_patch: non-conformable arrays
```

<sup>Created on 2019-01-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by mhesselbarth** [comment]
Is it just one patch or one cell? But in both cases it should be zero I guess...Didn't we already include this as a special case? Strange...

**Comment by Nowosad** [comment]
Just one patch. Yep, I also was surprised by this...

**Comment by Nowosad** [comment]
For now, I just added a test signalizing this error - https://github.com/r-spatialecology/landscapemetrics/commit/f4d11a9230906f18070e7d83a6dd0340a8c6791a

**Comment by marcosci** [comment]
But shouldn't it be only 0 if the patch is one cell? In Jakub case it should actually return something like 100* 100 - 400,  or?

**Comment by mhesselbarth** [comment]
Correct

**Comment by Nowosad** [comment]
@marcosci Can you explain that? Why 100 * 100 - 400, and not 100*100? What's the fragstats result in this kind of cases?

**Comment by marcosci** [comment]
Good question, could also be 100*100. Depends on what FRAGSTATS (or we) define as landscape boundary there.

Is the border of the raster the patch boundary, or landscape boundary in Suh a case?

**Comment by Nowosad** [comment]
I think it is a largely philosophical question (without just one convincing answer). I think we should follow FRAGSTATS is such cases..

**Comment by marcosci** [comment]
Just checked, FRAGSTATS returns a value 0f .9604 for your landscape.

**Comment by mhesselbarth** [comment]
Also if you check the box to not include the landscape boundary?

**Comment by marcosci** [comment]
You can't tick that on patch level, or do I miss one of the million buttons in fragstats?

**Comment by mhesselbarth** [comment]
I don't remeber. Will check when I'm in the office in a bit

**Comment by mhesselbarth** [comment]
Alright...should be fixed now (6ec4b2a3 ). Depending on the `consider_boundary` argument we either get core = 1 (because no edge is present) or core = 0.9604 (same as FRAGSTATS)

``` 
> lsm_p_core(x, consider_boundary = TRUE)
# A tibble: 1 x 6
  layer level class    id metric value
  <int> <chr> <int> <int> <chr>  <dbl>
1     1 patch     2     1 core       1

> lsm_p_core(x, consider_boundary = FALSE)
# A tibble: 1 x 6
  layer level class    id metric value
  <int> <chr> <int> <int> <chr>  <dbl>
1     1 patch     2     1 core   0.960
```

**Comment by mhesselbarth** [comment]
Please re-open if anyone is not happy with the solution
